src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -85,9 +84,58 @@ export default class SimidMapCreative extends BaseSimidCreative {
    * @private 
   */
   grantLocationAccess_() {
+    //ToDo(kristenmason@): implement the Google Maps request access functionality
+    findNearest.classList.add("hidden");
+    this.simidProtocol.sendMessage(CreativeMessage.REQUEST_PAUSE).catch(() => {
+      const pauseErrorMessage = {
+        message: "WARNING: Request to pause ad failed",
+      };
+      this.simidProtocol.sendMessage(CreativeMessage.LOG, pauseErrorMessage);
+    });
+    this.createMapState_();
```

Comment:
Your comments down in the tests say that the map will only load if pause is granted, but it looks like this will create it either way?

Reply:
Great catch! I should of called createMapState_() only if pauseRequest is resolved. Fixed